src/main/java/graphql/nadel/engine/HydrationInputResolver.java

```diff
@@ -266,23 +266,24 @@ private boolean isBatchHydrationField(HydrationInputNode hydrationInputNode) {
                         isSyntheticHydration
                 );
 
-
-        CompletableFuture<RootExecutionResultNode> serviceResult = serviceExecutor
-                .execute(executionContext, queryTransformationResult, service, operation,
-                        serviceContexts.get(service), true);
-
-        return serviceResult
-                .thenApply(resultNode -> convertSingleHydrationResultIntoOverallResult(executionContext.getExecutionId(),
-                        hydrationInputNode,
-                        hydrationTransformation,
-                        resultNode,
-                        hydrationInputNode.getNormalizedField(),
-                        queryTransformationResult,
-                        getNadelContext(executionContext),
-                        resultComplexityAggregator
-                ))
-                .whenComplete(this::possiblyLogException);
-
+        return queryTransformationResultCF.thenCompose(queryTransformationResult -> {
```

Comment:
Just saying, these diffs suck. All that's changed here is two lines added and some indentation.